src/utils/compound.js

```diff
@@ -0,0 +1,149 @@
+import { getPassiveUserRewardsData, getVotingPower } from "./rewards";
+
+export const calculateOptimalCompoundInterestWithFees = async ({
+  msDelta,
+  getMaxDate,
+  lockedOceanAmount,
+  formAmount,
+  formUnlockDate,
+  fees,
+  totalVeOceanSupply,
+}) => {
+  let optimalCompounds = 1;
+  let currentCompoundCount = 1;
+  let highestNetReturn = 0;
+  const principal = formAmount + lockedOceanAmount;
+  let optimalCompoundDetails = [];
+  // Calculate the optimal number of compounds
+  // the 4000 is just a random number, we will never reach it
+  let reachedHighestNetReturn = false;
+  let lastCompoundDetails = null;
+  while (!reachedHighestNetReturn) {
+    let totalRewards = 0;
+    let currentPrincipal = principal;
+    let tempTotalSupply = totalVeOceanSupply;
+    const compoundDetails = [];
+    for (let i = 0; i < currentCompoundCount; i++) {
+      const periodMS = msDelta / currentCompoundCount;
+      
+      const votingPower = getVotingPower(
+        periodMS,
+        getMaxDate,
+        currentPrincipal
+      );
+
+      const rewardsData = await getPassiveUserRewardsData(
+        votingPower,
+        currentPrincipal,
+        tempTotalSupply,
+        1
+      );
+
+      let periodReward = rewardsData.rewards;
+      const {
+        totalFeeForPeriod,
+        feeForClaims,
+        feeUpdateAmount,
+        mandatoryClaimCount,
+        totalPeriodClaims,
+      } = calculateFeeForPeriod(fees, periodMS);
+
+      const rawPeriodReward = periodReward;
+
+      if (i === 0) {
+        periodReward -= fees.lock;
+      }
+
+      if (i === currentCompoundCount - 1) {
+        periodReward -= totalPeriodClaims;
+        periodReward -= fees.withdraw;
```

Comment:
I don't think we need to subtract the lock and withdraw fees from specific periods. Those should not impact the period rewards but only the entire lock rewards.